src/com/owncloud/android/ui/dialog/RemoveFilesDialogFragment.java

```diff
@@ -109,37 +112,58 @@ public static RemoveFilesDialogFragment newInstance(OCFile file) {
         return newInstance(list);
     }
 
-    
+
     @Override
     public Dialog onCreateDialog(Bundle savedInstanceState) {
         Dialog dialog = super.onCreateDialog(savedInstanceState);
         mTargetFiles = getArguments().getParcelableArrayList(ARG_TARGET_FILES);
-        
+
         setOnConfirmationListener(this);
-        
+
         return dialog;
-    }    
+    }
 
     /**
      * Performs the removal of the target file, both locally and in the server.
      */
     @Override
     public void onConfirmation(String callerTag) {
         ComponentsGetter cg = (ComponentsGetter) getActivity();
+
+        stopMediaFiles(cg);
+
         cg.getFileOperationsHelper().removeFiles(mTargetFiles, false);
     }
-    
+
     /**
      * Performs the removal of the local copy of the target file
      */
     @Override
     public void onCancel(String callerTag) {
         ComponentsGetter cg = (ComponentsGetter) getActivity();
+
+        stopMediaFiles(cg);
+
         cg.getFileOperationsHelper().removeFiles(mTargetFiles, true);
     }
 
     @Override
     public void onNeutral(String callerTag) {
         // nothing to do here
     }
+
+    private void stopMediaFiles(ComponentsGetter cg) {
+        for (OCFile ocFile : mTargetFiles) {
+            FileDataStorageManager storageManager = cg.getStorageManager();
+            OCFile file = storageManager.getFileById(ocFile.getFileId());
```

Comment:
There is no need to retrieve a 'fresh' copy of the file instances from the storage manager to get the ```MediaService``` . We should remove it and avoid that way some extra reads from the database.